pkg/wasm/cache.go

```diff
@@ -409,36 +408,26 @@ func (c *LocalFileCache) getEntry(key cacheKey, ignoreResourceVersion bool) (str
 	return modulePath, key.checksum
 }
 
-// Purge periodically clean up the stale Wasm modules local file and the cache map.
+// purge clean up the stale Wasm modules local file and the cache map.
 func (c *LocalFileCache) purge() {
-	ticker := time.NewTicker(c.PurgeInterval)
-	defer ticker.Stop()
-	for {
-		select {
-		case <-ticker.C:
-			c.mux.Lock()
-			for k, m := range c.modules {
-				if !m.expired(c.ModuleExpiry) {
-					continue
-				}
-				// The module has not be touched for expiry duration, delete it from the map as well as the local dir.
-				if err := os.Remove(m.modulePath); err != nil {
-					wasmLog.Errorf("failed to purge Wasm module %v: %v", m.modulePath, err)
-				} else {
-					for downloadURL := range m.referencingURLs {
-						delete(c.checksums, downloadURL)
-					}
-					delete(c.modules, k)
-					wasmLog.Debugf("successfully removed stale Wasm module %v", m.modulePath)
-				}
+	c.mux.Lock()
+	for k, m := range c.modules {
+		if !m.expired(c.ModuleExpiry) {
```

Comment:
what will happen, if there's no xds update in 24h?

Reply:
IIRC, if ecds is cached in envoy, reconnect will not send ECDS request to xdsserver?

Reply:
If no ECDS, no purgin will happen.

`purge` function is only called at the end of `Get` function.

Reply:
> IIRC, if ecds is cached in envoy, reconnect will not send ECDS request to xdsserver?

Yes. ECDS request will be not sent to xdsserver if Envoy already has proper resources.